src/multipart.cpp

```diff
@@ -0,0 +1,51 @@
+#include "persistence.hpp"
+#include "multipart.hpp"
+#include <cctype>
+#include <sstream>
+#include <fmt/format.h>
+
+// The way that multipart is going to work here is that they will exist as directories(collections fine)
+// in the irods server under a temporary directory.
+//
+// The path of the finished object needs to be able to be encoded to a data-object name safe string.
+// It does, however, not need to be decoded from the other end.
+
+std::string temporary_directory = "/tempZone/temp";
```

Comment:
- `temporary_directory` -> `temporary_collection`
- Move into an anonymous namespace?
- Perhaps make it can be changed to a `const std::string_view` or `const char* const`